common/persistence/persistence_interface.go

```diff
@@ -232,8 +232,8 @@ type (
 	// create the shard with the returned value.
 	InternalGetOrCreateShardRequest struct {
 		ShardID          int32
-		CreateShardInfo  func() (rangeID int64, shardInfo *commonpb.DataBlob, err error)
-		LifecycleContext context.Context // cancelled when shard is unloaded
+		CreateShardInfo  func() (rangeID int64, shardInfo *commonpb.DataBlob, err error) `json:"-"`
```

Comment:
Cannot be serialized as JSON.

Reply:
I think you should add a comment to this file explaining that this struct needs to be JSON serializable for tracing to work. And that those JSON settings is just for that.